domain_models/models.py

```diff
@@ -211,3 +212,11 @@ def get_data(self):
         return dict((name, field.get_builtin_type(self))
                     for name, field in
                     six.iteritems(self.__class__.__fields__))
+
+    def set_data(self, data):
+        """Set dictionary data to model.
+
+        :param dict data:
+        """
+        for name, value in six.iteritems(data):
+            self.__class__.__fields__[name].set_builtin_type(self, value)
```

Comment:
Wait... It will work the same in both cases or I strongly miss something... Also, ``Field.set_value()`` already works with required fields...